SpatialGDK/Source/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -0,0 +1,230 @@
+				RemoteFunction->FunctionFlags & FUNC_NetMulticast)
+			{
+				ERPCType RPCType;
+				if (RemoteFunction->FunctionFlags & FUNC_NetClient)
+				{
+					RPCType = RPC_Client;
+				}
+				else if (RemoteFunction->FunctionFlags & FUNC_NetServer)
+				{
+					RPCType = RPC_Server;
+				}
+				else if (RemoteFunction->FunctionFlags & FUNC_NetCrossServer)
+				{
+					RPCType = RPC_CrossServer;
+				}
+				else if (RemoteFunction->FunctionFlags & FUNC_NetMulticast)
+				{
+					RPCType = RPC_NetMulticast;
+				}
+				else
+				{
+					RPCType = RPC_Count;
+					checkNoEntry();
+				}
+
+				TArray<UFunction*>& RPCArray = Info.RPCs.FindOrAdd(RPCType);
+
+				FRPCInfo RPCInfo;
+				RPCInfo.Type = RPCType;
+				RPCInfo.Index = RPCArray.Num();
+
+				RPCArray.Add(*RemoteFunction);
+				Info.RPCInfoMap.Add(*RemoteFunction, RPCInfo);
+			}
+		}
+
+		for (TFieldIterator<UProperty> PropertyIt(Class); PropertyIt; ++PropertyIt)
+		{
+			UProperty* Property = *PropertyIt;
+
+			if (Property->PropertyFlags & CPF_Handover)
+			{
+				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
+				{
+					FHandoverPropertyInfo HandoverInfo;
+					HandoverInfo.Handle = Info.HandoverProperties.Num() + 1; // 1-based index
+					HandoverInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
+					HandoverInfo.ArrayIdx = ArrayIdx;
+					HandoverInfo.Property = Property;
+
+					Info.HandoverProperties.Add(HandoverInfo);
+				}
+			}
+		}
+		
+		// TODO: Probably clean up duplication?
+		Info.SingleClientComponent = SchemaDatabase->ClassToSchema[Class].SingleClientRepData;
+		ComponentToClassMap.Add(Info.SingleClientComponent, Class);
+
+		Info.MultiClientComponent = SchemaDatabase->ClassToSchema[Class].MultiClientRepData;
+		ComponentToClassMap.Add(Info.MultiClientComponent, Class);
+
+		Info.HandoverComponent = SchemaDatabase->ClassToSchema[Class].HandoverData;
+		ComponentToClassMap.Add(Info.HandoverComponent, Class);
+
+		Info.RPCComponents[RPC_Client] = SchemaDatabase->ClassToSchema[Class].ClientRPCs;
+		ComponentToClassMap.Add(Info.RPCComponents[RPC_Client], Class);
+
+		Info.RPCComponents[RPC_Server] = SchemaDatabase->ClassToSchema[Class].ServerRPCs;
+		ComponentToClassMap.Add(Info.RPCComponents[RPC_Server], Class);
+
+		Info.RPCComponents[RPC_NetMulticast] = SchemaDatabase->ClassToSchema[Class].NetMulticastRPCs;
+		ComponentToClassMap.Add(Info.RPCComponents[RPC_NetMulticast], Class);
+
+		Info.RPCComponents[RPC_CrossServer] = SchemaDatabase->ClassToSchema[Class].CrossServerRPCs;
+		ComponentToClassMap.Add(Info.RPCComponents[RPC_CrossServer], Class);
+
+		if (Class->IsChildOf<AActor>())
+		{
+			if (AActor* ContainerCDO = Cast<AActor>(Class->GetDefaultObject()))
+			{
+				TInlineComponentArray<UActorComponent*> NativeComponents;
+				ContainerCDO->GetComponents(NativeComponents);
+
+				for (UActorComponent* Component : NativeComponents)
+				{
+					if (IsSupportedClass(Component->GetClass()))
```

Comment:
Lets make a function that verifies the class is supported as well as ensuring the class doesn't already exist to warn against duplicate component types. Unless that check happens elsewhere?